File: tarea5/polinomio.c
#include <stdio.h>

int main(int argc, char *argv[]) {
	
	int c [2];
	int e [2];
	int c1 [2];	int e1 [2];
	int suma=0;
	int i;
	int j;
	int suma2;
	
	printf("deme el primer polinomio:\n");
	for (i=0;i<2;i++)
	{
		
		printf("deme el coeficiente:%d\n",i+1);		scanf("%d",&c[i]);	printf("deme el exponente:%d\n",i+1);	scanf("%d",&e[i]);
	
	
	
    }
	printf("deme el segundo polinomio\n");
	for (j=0;j<2;j++)	{
		printf("deme el coeficiente:%d \n",j+1);		scanf("%d",&c1[j]);		
		
		printf("deme el exponente:%d\n",j+1);		scanf("%d",&e1[j]);		
		
	}
	
	if ( e[0]== e1[0])
	{
		suma= c1[0]+ c[0];
	}
	
	if ( e[1] == e1[1])	{		suma2= c1[1]+ c[1];	}
	
	
	printf("%dx*%d + %d",suma,e[0],suma2);
	
	return 0;
}

File: Evaluacion2/SumaPolinCyScalaArray/polinomio.scala
import scala.util.control._

object polinomioN{
  def main(args: Array[String]){
    var may = 0
    var men = 0
    val loop = new Breaks;
    println("Deme el numero de terminos del primer polinomio:")
      val n = readInt()
      var polin = new Array[Int](n)
      var grado = new Array[Int](n)

      for(i <- 0 to n-1) {
         println("Deme el coeficiente del termino " + i)
         polin(i) = readInt()
         println("Deme el grado del termino "+i)
         grado(i) = readInt()
       }

    println("Deme el numero de terminos del segundo polinomio:")
      val m = readInt()
      var polin2 = new Array[Int](m)
      var grado2 = new Array[Int](m)

     for(i <- 0 to m-1) {
        println("Deme el coeficiente del termino " + i)
        polin2(i) = readInt()
        println("Deme el grado del termino "+i)
        grado2(i) = readInt()
      }
     if(n > m){
       may = n
     }else{
       may = m
     }
     var result = new Array[Int](may)
     loop.breakable{
     for(i <- 0 to may-1){
       if(grado(i) == grado2(i)){
         result(i) = polin(i) + polin2(i)
       }else{
         loop.break;
       }
     }
   }
     for(i <- 0 to may-1){
       println(result(i))
     }
  }
}
